File: projects/m1/040-password-validation/solutions/andreaattardo/passwordvalidation.cpp
/*
Password validation
	In this exercise you will write a function that determines whether or not a password is good.We will define a good 
	password to be a one that is at least 8 characters longand contains at least one uppercase letter, at least one 
	lowercase letter, and at least one number.Your function should return True if the password passed to it as its only 
	parameter is good.Otherwise it should return False.Include a main program that reads a password from the user and 
	reports whether or not it is good.Ensure that your main program only runs when your solution has not been imported into
	another file.
*/

#include <iostream>
#include <cctype>
#include <string>
using namespace std;

bool isGoodPsw(const string& psw) {

	//check psw lenght < 8
	int pswLen = psw.length();
	if (pswLen < 8) return false;

	//check at least 1 lowercase, 1 uppercase, 1 number
	int	 counter =	0b000; //binary number, 0b111 is all 3 conditions true

	for (int i = 0; counter != 0b111 && i < pswLen; i++) {

		if ( ( counter < 0b001 ) && islower( psw[i]) )	counter +=   0b1;
		if ( ( counter < 0b010 ) && isupper( psw[i]) )	counter +=  0b10;
		if ( ( counter < 0b100 ) && isdigit( psw[i]) )	counter += 0b100;

	}

	if (counter == 0b111) return true;
	else return false;
}


int main() {

	string	usrPsw;

	cout << "Insert a password that is at least " 
		 << "8 characters with 1 uppercase, " 
		 << "1 lowercase and 1 number." 
		 << endl;
	getline(cin, usrPsw);

	if (isGoodPsw(usrPsw)) cout << "It's a good password";
	else cout << "No good password";

	return 0;
}

File: projects/m1/040-password-validation/solutions/M.GIOVARRUSCIO/PASSWORD-VALIDATION.py
"""
In this exercise you will write a function that determines whether or not a password is good.
We will define a good password to be a one that is at least 8 characters long and contains
at least one uppercase letter,
at least one lowercase letter,
and at least one number.
Your function should return True if the password passed to it as its only parameter is good.
Otherwise it should return False.
Include a main program that reads a password from the user and reports whether or not it is good.
Ensure that your main program only runs when your solution has not been imported into
another file.

"""
lista=['0','1','2','3','4','5','6','7','8','9']
def controlla(st):
    #least 8 characters long
    if len(s)>8:
        maiuscola=0
        minuscola=0
        for i in s:
            if i.isupper():
                maiuscola=1
            else:
                minuscola=1
        #at least one uppercase letter,
        #at least one lowercase letter,
        if maiuscola==1 and minuscola==1:
            num=0
            #and at least one number
            for i in s:
                if i in lista:
                    num=1
            if num>0:
                return True
            else:
                return False
        else:
            return False

    else:
        return False
if __name__ == '__main__':
    s=input("Inserisci stringa:")
    if(controlla(s)):
        print("PASSWORD BUONA")
    else:
        print("PASSWORD NON BUONA")
